### PR TITLE
Add Parser for file /etc/dnf/dnf.conf

### PR DESCRIPTION
**Pull request by shlao**
Added
```
insights/parsers/dnf_conf.py
insights/parsers/tests/test_dnf_conf.py
docs/shared_parsers_catalog/dnf_conf.rst
```

Updated
```
insights/specs/__init__.py
insights/specs/default.py
```

Signed-off-by: shlao <shlao@redhat.com>

### CODE REVIEW

**Comment by bfahr** [comment]
Hold for resolution of whether this should be filtered based on [documentation](https://dnf.readthedocs.io/en/latest/conf_ref.html).

**Comment by patchkez** [comment]
test this please

File: insights/specs/__init__.py [hunk 1]
```diff
@@ -122,6 +122,7 @@ class Specs(SpecSet):
     dmesg_log = RegistryPoint(filterable=True)
     dmidecode = RegistryPoint()
     dmsetup_info = RegistryPoint()
+    dnf_conf = RegistryPoint()
```
Comment (bfahr) [inline insights/specs/__init__.py:125]: @shlao please change this spec to add `filterable=True`.

File: insights/parsers/dnf_conf.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    >>> dconf.has_option('main', 'foo')
+    False
+"""
+
+from insights import parser
+from insights.specs import Specs
+from insights.parsers.yum_conf import YumConf
+
+
+@parser(Specs.dnf_conf)
```
Comment (bfahr) [inline insights/parsers/dnf_conf.py:35]: You probably should add a filter to this module to ensure that the input is minimally parsable.  Something like `add_filter(Specs.dnf_conf, '[')` should be sufficient.

**Pull request merged**